deploy/charts/cert-manager/templates/deployment.yaml

```diff
@@ -137,6 +137,9 @@ spec:
           {{- with .Values.dns01RecursiveNameservers }}
           - --dns01-recursive-nameservers={{ . }}
           {{- end }}
+          {{- if .Values.disableAutoApproval }}
+          - --controllers=-certificaterequests-approver
+          {{- end }}
```

Comment:
suggestion: I think this could use manual testing to confirm that it does what's expected. Notably this looks different to how we set this in docs which is usually `--controllers=*\,-certificaterequests-approver'`

I'm not saying this is wrong, but I think it would be helpful to show that it's right in the PR description!

Reply:
Also, how does this interact with someone setting the --controllers flag through extraArgs in the chart?

Reply:
I was just testing this and there seems to be an issue still.

Reply:
I think what you propose is pretty unsurprising for a user, and it wouldn't be _bad_.  In a different place you said that our current logic is mirroring k8s - it'd be good to know if there's a reason they chose to do it the way they did.

I think an alternative approach (given that the only controller regularly disabled this way) is to add an entirely new flag, `--disable-default-approver`.

Obviously that's making a special case out of the default approver, but I think it _is_ a special case.

I'd envision that first we'd parse the `--controllers` flag, and then afterwards if `--disable-default-approver` is set, remove it from the list of controllers.

Reply:
The upstream implementation has not changed much since its introduction in 2017: https://github.com/kubernetes/kubernetes/pull/39740.
I don't think they specifically looked at this specific situation (a setup with only negative controller selectors).

Reply:
I implemented the proposed change in this commit: https://github.com/cert-manager/cert-manager/pull/7049/commits/985607b08cef948dfb628d29b30fe96c526a0160